.github/workflows/deployment.yml

```diff
@@ -11,24 +13,38 @@ on:
 
 jobs:
 
-  build:
+  Version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.daily-version.outputs.version }}
+      created: ${{ steps.daily-version.outputs.created }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: fregante/daily-version-action@v1
+        name: Create tag if necessary
+        id: daily-version
+
+  Build:
+    needs: Version
+    if: github.event_name == 'push' || needs.Version.outputs.created
```

Comment:
Correctly passes all tested situations:
- any `push` has a tag, either because the push itself is tag or it was created by the `Version` step
- the `created` output is correctly set only if HEAD wasn't initially tagged, in any situation